geode-core/src/test/java/org/apache/geode/internal/cache/wan/GatewayReceiverStatsTest.java

```diff
@@ -60,27 +63,50 @@ public void setup() {
     when(factory.createAtomicStatistics(any(), any()))
         .thenReturn(statistics);
 
-    registry = new CompositeMeterRegistry();
+    registry = new SimpleMeterRegistry();
 
   }
 
+  @After
+  public void closeStats() {
+    if (gatewayReceiverStats != null) {
+      gatewayReceiverStats.close();
+    }
+  }
+
   @Test
   public void incEventsReceived_incrementsTheEventsReceivedStat() {
     int eventsReceivedStatId = 33;
 
     when(statisticsType.nameToId(EVENTS_RECEIVED_STAT_NAME))
         .thenReturn(eventsReceivedStatId);
 
-    GatewayReceiverStats gatewayReceiverStats =
-        GatewayReceiverStats.createGatewayReceiverStats(factory, ownerName, registry);
+    gatewayReceiverStats = createGatewayReceiverStats(factory, ownerName, registry);
 
     int delta = 99;
 
     gatewayReceiverStats.incEventsReceived(delta);
 
     verify(statistics).incInt(eventsReceivedStatId, delta);
+  }
 
-    gatewayReceiverStats.close();
+  @Test
+  public void incEventsReceived_incrementsTheRegisteredEventsReceivedCounter() {
```

Comment:
The first 5 tests in this class seem to be testing the same things that are already tested in LegacyStatCounterTest, i.e. making sure increment increments both the stat and the counter, that the counter returns the value from the stat, and that the description and units on the counter match the stat. Rather than testing these same things for each LegacyStatCounter we add, I would rather have these tests live in LegacyStatCounterTest.

Reply:
I do want to ensure that `GatewayReceiverStats` connects its `incEventsReceived()` and `getEventsReceived()` methods to the appropriate stat, which is something `LegacyStatCounter` can test on its own. The tests are similar, but there's a key difference in the responsibilities being tested: The `LegacyStatCounter` tests test that the counter connects to whatever stat you tell it to connect to. The `GatewayReceiverStats` tests test that these particular `GatewayReceiverStats` methods connect specifically to the "events received" stat.